public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -262,7 +271,7 @@ export function getNewDashboardModelData(urlFolderId?: string, panelType?: strin
       isNew: true,
       folderId: 0,
     },
-    dashboard: {
+    dashboard: pendingNewDashboard.dashboard ?? {
```

Comment:
you could have used setDashboardToFetchFromLocalStorage and checked DASHBOARD_FROM_LS_KEY, the benefit is then the Preview button could open the debug dashboard in a new tab (in case you did not want to lose the state you had in the dashboard you where troubleshooting) 

DASHBOARD_FROM_LS_KEY / localstorage trick is used from Explore to open a specific dashboard with some unsaved modifications applied. The same process could be used to create a new dashboard with some pre defined panels (exactly like pendingNewDashboard).